### PR TITLE
Panzer: Add Tpetra tests for disc-fe/la_factory

### PR DESCRIPTION
**Pull request by thearusable**
This PR adds Tpetra version of tests in disc-fe/la_factory which were previously implemented for Epetra.

### CODE REVIEW

**Comment by thearusable** [comment]
@rppawlo @JacobDomagala

File: packages/panzer/disc-fe/test/la_factory/CMakeLists.txt [hunk 1]
```diff
@@ -42,31 +42,47 @@ INCLUDE_DIRECTORIES(REQUIRED_DURING_INSTALLATION_TESTING
                     ${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_CURRENT_SOURCE_DIR}/../../../dof-mgr/test/dofmngr_test)
 
+SET(FILES tTpetraLinearObjFactory.cpp tTpetra_GlbEvalData.cpp UnitTest_GlobalIndexer.cpp)
+IF(PANZER_HAVE_EPETRA)
+  APPEND_SET(FILES tBlockedLinearObjFactory.cpp tBlockedTpetraLinearObjFactory.cpp
+             tEpetra_GlbEvalData.cpp)
+ENDIF(PANZER_HAVE_EPETRA)
+
+IF(PANZER_HAVE_EPETRA)
```
Comment (JacobDomagala) [inline packages/panzer/disc-fe/test/la_factory/CMakeLists.txt:51]: No need to END and then BEGIN the same 'IF(PANZER_HAVE_EPETRA)'

Comment (thearusable) [reply packages/panzer/disc-fe/test/la_factory/CMakeLists.txt:51]: Good point. I'll update that

**Review by rppawlo — DISMISSED** [review]

**Comment by thearusable** [comment]
@rppawlo Sorry, I pushed a small correction right after you approved this PR.

**Review by rppawlo — DISMISSED** [review]

**Comment by thearusable** [comment]
@rppawlo PR rebased after #10519

**Pull request merged**